Source/WebKit/WebProcess/cocoa/WebProcessCocoa.mm

```diff
@@ -862,24 +862,33 @@ static void prewarmLogs()
     if (!connectionPair)
         CRASH();
     auto [streamConnection, serverHandle] = WTFMove(*connectionPair);
-    m_logStreamConnection = WTFMove(streamConnection);
-    if (RefPtr logStreamConnection = m_logStreamConnection)
-        logStreamConnection->open(*this);
 
-    parentProcessConnection()->sendWithAsyncReply(Messages::WebProcessProxy::SetupLogStream(getpid(), WTFMove(serverHandle), WebProcess::singleton().m_logStreamIdentifier), [] (IPC::Semaphore&& wakeUpSemaphore, IPC::Semaphore&& clientWaitSemaphore) {
-        if (RefPtr logStreamConnection = WebProcess::singleton().m_logStreamConnection)
-            logStreamConnection->setSemaphores(WTFMove(wakeUpSemaphore), WTFMove(clientWaitSemaphore));
+    LogStreamIdentifier logStreamIdentifier { LogStreamIdentifier::generate() };
+
+    RefPtr logStreamConnection = WTFMove(streamConnection);
+    if (!logStreamConnection)
+        return;
+
+    logStreamConnection->open(*this, RunLoop::protectedCurrent());
+
+    parentProcessConnection()->sendWithAsyncReply(Messages::WebProcessProxy::SetupLogStream(getpid(), WTFMove(serverHandle), logStreamIdentifier), [logStreamConnection, logStreamIdentifier] (IPC::Semaphore&& wakeUpSemaphore, IPC::Semaphore&& clientWaitSemaphore) {
+        logStreamConnection->setSemaphores(WTFMove(wakeUpSemaphore), WTFMove(clientWaitSemaphore));
 #if PLATFORM(IOS_FAMILY)
         prewarmLogs();
 #endif
+        RefPtr logClient = WebKitLogClient::create(*logStreamConnection, logStreamIdentifier);
+        webkitLogClient() = logClient;
+        webCoreLogClient() = logClient;
```

Comment:
Still unclear why we need 2 clients (and ref-counting)

Reply:
It is a very good point. I am looking into if we can have a single log client, and still have a separation between WebCore and WebKit log definitions.

Reply:
And then I'd rename `webCoreLogClient()` into `logClient()`

Reply:
That's a very good idea! I was looking into solving this by how we generate the declarations, but down casting seems a much more elegant approach :)